ui/puzzle/src/ctrl.ts

```diff
@@ -463,6 +467,75 @@ export default class PuzzleCtrl implements ParentCtrl {
+  private reportIfMultipleSolutions = (ev: Tree.ClientEval): void => {
+    // if the eval depth is superior to 20, we're on the puzzle solution and two moves are good
+    // that is an absolute eval superior to 4, then log it
+    console.log('ev', ev, 'node', this.node);
+    // first, make sure we're in view mode so we know the solution is the mainline
+    // do not check, checkmate puzzles
+    if (
+      !this.session.userId ||
+      this.reportedForMultipleSolutions ||
+      this.mode != 'view' ||
+      this.threatMode() ||
+      // the `mate` key theme is not sent, as it is considered redubant with `mateInX`
+      this.data.puzzle.themes.some((t: ThemeKey) => t.toLowerCase().includes('mate'))
+    )
+      return;
+    console.log('good mode');
+    const node = this.node;
+    // more resilient than checking the turn directly, if eventually puzzles get generated from 'from position' games
+    const nodeTurn = node.fen.includes(' w ') ? 'white' : 'black';
+    if (
+      node.ply >= this.initialNode.ply &&
+      nodeTurn == this.pov &&
+      this.mainline.some(n => n.id == node.id)
+    ) {
+      console.log('correct position!');
+      const invertIfBlack = (cp: number) => (this.pov == 'white' ? cp : -cp);
+      console.log('ev.depth > 20', ev!.depth > 20, 'ev.depth > 20');
+      if (ev.pvs[1]?.cp) {
+        console.log(
+          'ev!.pvs[1]!.cp',
+          ev.pvs[1].cp,
+          'ev.pvs[1].cp  >= cpThreshold',
+          invertIfBlack(ev.pvs[1].cp) >= 400,
+        );
+      }
+      // TODO probably check what are really the conditions for a puzzle to have multiple solutions
+      if (ev.depth > 20 && ev.pvs[1]?.cp && invertIfBlack(ev.pvs[1].cp) >= 400) {
+          this.reportedForMultipleSolutions = true;
+      //   //this.reportDialog()
+        const reason = `after move ${node.ply}${node.san}, at depth ${ev.depth}, they're multiple solutions, pvs ${ev.pvs.map(pv => `${pv.moves[0]}: ${pv.cp}`).join(', ')}`;
+        xhr.report(this.data.puzzle.id, reason)
+      }
+    }
+  };
+
+  // TODO FIXME, does not work, dialogs...
+  // private reportDialog = () => {
+  //   // html form yes/no report with a checkbox 'do not show this again for a week'
+  //   domDialog({
+  //     show: 'modal',
+  //     htmlText:
+  //       '<div><strong style="font-size:1.5em">' +
+  //       'Report multiple solutions' +
+  //       '</strong><br /><br />' +
+  //       '<pre>' +
+  //       'You have found a puzzle with multiple solutions, report it?' +
+  //       '</pre><br />' +
+  //       '<br /><br />' +
+  //       `<button type="reset" class="button button-empty button-red text reset" data-icon="${licon.X}">No</button>` +
+  //       `<button type="submit" class="button button-green text apply" data-icon="${licon.Checkmark}">Yes</button>`,
+  //   }).then(_dlg => {
+  //     // No overload matches this call.
+  //     // $('.reset', dlg.view).on('click', dlg.close());
+  //     // dlg.showModal();
+  //   });
+  // };
```

Comment:
cc @schlawg, I was not sure what type of dialog would be best here, dom or snab. I also could not get it working 😅 

```
$('.reset', dlg.view).on('click', dlg.close());
 ```
 was returning ts error `No overload matches this call.` but it's used elsewhere like in `diagnosticDialog.ts`
 
 ```
  $('.err', dlg.view).on('focus', select);
  $('.clear', dlg.view).on('click', () => log.clear().then(() => dlg.close()));
  $('.copy', dlg.view).on('click', () =>
  ```

Reply:
Oh well I was tired yesterday, ` $('.reset', dlg.view).on('click', () => dlg.close());` fixed it.

Generally this part of the code feels hacky, I'd like some help if possible for styling and adding the `Hide this popup for a week` toggle in the dialog, stored in local storage.